packages/core/useVirtualList/index.ts

```diff
@@ -16,10 +17,17 @@ export interface UseVirtualListOptions {
   overscan?: number
 }
 
-export function useVirtualList <T = any>(list: T[], options: UseVirtualListOptions) {
+export type UseVirtualListItem<T> = {
+  data: T
+  index: number
+}
+
+export function useVirtualList <T = any>(list: MaybeRef<T[]>, options: UseVirtualListOptions) {
   const containerRef: Ref = ref<HTMLElement | null>()
   const size = useElementSize(containerRef)
-  const currentList: Ref = ref([])
+
+  const currentList: Ref<UseVirtualListItem<T>[]> = ref([])
+  const _list = ref(isRef(list) ? list.value : list)
```

Comment:
```suggestion
  const _list = ref(list)
```

Vue will check it's already a ref or not and reuse the existing one if possible

Reply:
Nice, ok. Doing this locally. Do you have any suggestions for how to fix the types? I'm struggling a bit to understand why `_list.value` can't be assigned to `currentList.value`.

<img width="530" alt="Screen Shot 2021-11-05 at 6 03 07 PM" src="https://user-images.githubusercontent.com/2801156/140583522-256fc1fe-ed4b-4887-88ec-92ef7e641778.png">